woocommerce/woocommerce-gateway-stripe
# Fix Stripe disable modal text

When Link is enabled, it is present in the `enabled_payment_method_ids` in the [response](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/admin/class-wc-rest-stripe-settings-controller.php#L228) to the `settings` data call. [This function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-stripe-helper.php#L757-L761) returns the data which are used in the Stripe settings page.

As a side effect of Link being present in the `enabled_payment_method_ids` list, the [disable confirmation modal](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/client/entrypoints/payment-gateways/disable-confirmation-modal.js) shows wrong text.

## Changes proposed in this Pull Request:
- Excluding link from `enabled_payment_method_ids` as an express checkout method.

## Testing instructions
- Enable Stripe.
- Enable ECE feature flag (return `true` from the `is_stripe_ece_enabled` function or set the `_wcstripe_feature_ece` option as `yes`)
- Disable all payment methods except Google/Apple Pay and Link.
- Go to **WooCommcerce > Settings > Payments**
- Toggle Stripe to disable the gateway. You should see a confirmation modal.
- In the `develop` branch, the modal shows [the message](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/client/entrypoints/payment-gateways/disable-confirmation-modal.js#L26) intended for the scenario when other payment methods are enabled.

<img width="653" alt="Screenshot 2024-10-21 at 2 34 04 PM" src="https://github.com/user-attachments/assets/59968fa0-f94c-44ef-bab3-8f327afcffe2">

- In this branch, the modal should display [the message](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/client/entrypoints/payment-gateways/disable-confirmation-modal.js#L22) intended for the scenario when only express methods are enabled.
<img width="653" alt="Screenshot 2024-10-21 at 3 30 19 PM" src="https://github.com/user-attachments/assets/164914c6-d274-4297-95fc-eac007bb29c8">

- Test Link payment method to make sure there is no regression.
